### PR TITLE
Create pin manager

### PR DESCRIPTION
**Pull request by rockboynton**
# Date

Initial PR Date: 01/16/2020

Due Date: N/A

# Reviewers
 @anthony-bartman @Ivendrakeo 

# Description

This adds the basic pin manager to the hobby hub. For now it doesn't control much, the actual pin management is left to the Web UI. This creates, updates and maintains the pin configuration JSON file. It implements all of the features outlined in the Issue #15 

# Project Issue

# Test Results

Outline to test: 
with JSON file open:
    

1. request pin with tag 1 (unused)
2. request pin with tag 2 (unused)
3. get pin with tag 1 (used)
4. clear unused pins
5. verify tag 1 is still entered 
6. reset config 
7. verify config file is empty

### CODE REVIEW

**Review by anthony-bartman — APPROVED** [review]
Solid!

**Comment by Ivendrakeo** [comment]
I like a lot of the backend code. I think what you have here is a solid foundation. However I guess I wasn't specific on what I meant by "write and API" in the issue description. The current interfacing method is via command line / shell. I guess this is fine for calling it from the web ui, however i'm not too sure why the web UI would need this since it could just read the whole JSON file? 

How does the person writing the user programs interface with these methods? I'm working on a python module to add our code, maybe this is best put in our module such that the user could import the code  and call high level methods directly?

File: scripts/init.sh [hunk 1]
```diff
@@ -9,3 +9,6 @@ cp files/init/start.sh /etc/hobby-hub/
 # Program manager
 mkdir -p /etc/hobby-hub/out/ # contains output of program manager
 cp files/src/program_manager.py /etc/hobby-hub/ # program manager itself
+
+# Pin mapping configuration
+echo "{}" | sudo tee -a /etc/hobby-hub/pin_mapping.json # creates empty, but valid JSON file
```
Comment (Ivendrakeo) [inline scripts/init.sh:14]: You've created the json file here but haven't copied the pin_manager.py . This is fine assuming you plan to add this to the python module. in which case we'll have to figure out where to place this later...

**Comment by rockboynton** [comment]
> I like a lot of the backend code. I think what you have here is a solid foundation. However I guess I wasn't specific on what I meant by "write and API" in the issue description. The current interfacing method is via command line / shell. I guess this is fine for calling it from the web ui, however i'm not too sure why the web UI would need this since it could just read the whole JSON file?
> 
> How does the person writing the user programs interface with these methods? I'm working on a python module to add our code, maybe this is best put in our module such that the user could import the code and call high level methods directly?

That's a good question and one I want to talk about with you and Barty. We have to determine what the will the user be doing through the Web UI and what they will be doing in the actual programs they write. Writing use cases and user stories would probably be helpful for that.

File: scripts/init.sh (hunk 1 above)
Comment (rockboynton) [reply scripts/init.sh:14]: Done. I added it the same way as the program manager

**Pull request merged**